pexpect/pty_spawn.py

```diff
@@ -752,10 +752,14 @@ def interact(self, escape_character=chr(29),
         child process in interact mode is duplicated to the given log.
 
         You may pass in optional input and output filter functions. These
-        functions should take a string and return a string. The output_filter
-        will be passed all the output from the child process. The input_filter
-        will be passed all the keyboard input from the user. The input_filter
-        is run BEFORE the check for the escape_character.
+        functions should take bytes array and return bytes array too. Even
+        after Python 3 ``unicode='utf-8'`` support, meth:`interact` doesn't
+        guarantee to pass input_filter and output_filter encoded str value.
```

Comment:
"doesn't guarantee" sounds like it can vary, which isn't the case: it always passes bytes.

Also, "encoded str" isn't quite right. A Python 3 str is *decoded*. Decoding converts bytes to str, encoding converts str to bytes.

Reply:
Will fix